### PR TITLE
Paging does not work as expected.

### PR DESCRIPTION
**Pull request by mglaman**
"Next" would re-fetch page `0`, then next click would work as expected.

### CODE REVIEW

File: js/src/components/app.js [hunk 1]
```diff
@@ -62,14 +62,16 @@ export default class App extends Component {
   }
   nextPageHandler = (e) => {
     e.preventDefault();
-    this.fetchLogEntries(this.state.page + 1);
+    this.state.page++;
```
Comment (dawehner) [inline js/src/components/app.js]: Do you mind using [this.setState](https://reactjs.org/docs/state-and-lifecycle.html) here instead?

**Comment by dawehner** [comment]
Nice fix!

File: js/src/components/app.js (hunk 1 above)
Comment (mglaman) [reply js/src/components/app.js]: Yeah, I just caught this myself.

**Comment by mglaman** [comment]
I think that should do it. Was a blind update. Reviewing my own code for bad state management.

File: js/src/components/app.js [hunk 2]
```diff
@@ -62,14 +62,20 @@ export default class App extends Component {
   }
   nextPageHandler = (e) => {
     e.preventDefault();
-    this.fetchLogEntries(this.state.page + 1);
+    this.setState({
+      page: this.state.page + 1
```
Comment (mglaman) [inline js/src/components/app.js:66]: Does not work, because setting the state is not exactly immediate

Comment (dawehner) [reply js/src/components/app.js:66]: In this case you can use ```this.setState((prevState, props) => ({
  page: prevState.page + 1,
}));```

Comment (mglaman) [reply js/src/components/app.js:66]: I didn't try that, but one issue I had was the async nature. It executed the HTTP request before state settled. I updated to

```
    this.setState({
      page: this.state.page - 1
    }, () => {
      this.fetchLogEntries(this.state.page);
    });
```

Comment (dawehner) [reply js/src/components/app.js:66]: Oh i see. Yeah the above example is useful when the HTTP request would change the actual state.

**Pull request merged**